tests/test_noise_model.py

```diff
@@ -374,3 +431,169 @@ def test_repr(self):
             "eff_noise_rates=(0.1,), eff_noise_opers=(((1, 0, 0), (0, 1, 0), "
             "(0, 0, 1)),), with_leakage=True)"
         )
+
+        assert (
+            repr(
+                NoiseModel(
+                    temperature=15.0,
+                    trap_depth=150.0,  # same units as temperature
+                    trap_waist=1.0,
+                    runs=1,
+                    samples_per_run=1,
+                )
+            )
+            == "NoiseModel(noise_types=('doppler', 'register'), runs=1, "
+            "samples_per_run=1, temperature=15.0, trap_waist=1.0, "
+            "trap_depth=150.0)"
+        )
+
+
+@pytest.mark.parametrize(
+    "register2D",
+    [
+        True,
+        False,
+    ],
+)
+def test_noisy_register(register2D) -> None:
+    """Testing noisy_register function in case 2D and 3D register."""
+    noise_model = MagicMock()
+    noise_model._register_sigma_xy_z.return_value = (0.13, 0.8)
+
+    if register2D:
+        qdict = {
+            "q0": np.array([-15.0, 0.0]),
+            "q1": np.array([-5.0, 0.0]),
+            "q2": np.array([5.0, 0.0]),
+            "q3": np.array([15.0, 0.0]),
+        }
+    else:
+        qdict = {
+            "q0": np.array([-15.0, 0.0, 0.0]),
+            "q1": np.array([-5.0, 0.0, 0.0]),
+            "q2": np.array([5.0, 0.0, 0.0]),
+            "q3": np.array([15.0, 0.0, 0.0]),
+        }
+
+    # Predefined deterministic noise
+    fake_normal_xy_noise = np.array(
+        [
+            [0.1, -0.1],
+            [0.2, -0.2],
+            [0.3, -0.3],
+            [0.5, -0.5],
+        ]
+    )
+    fake_normal_z_noise = np.array([0.05, 0.07, 0.09, 0.11])
+    with patch("numpy.random.normal") as mock_normal:
+        # moke the noise generation
+        mock_normal.side_effect = [
+            fake_normal_xy_noise,
+            fake_normal_z_noise,
+        ]
+        result = _noisy_register(qdict, noise_model)
+    expected_positions = {
+        "q0": np.array([-15.0 + 0.1, 0.0 - 0.1, 0.05]),
+        "q1": np.array([-5.0 + 0.2, 0.0 - 0.2, 0.07]),
+        "q2": np.array([5.0 + 0.3, 0.0 - 0.3, 0.09]),
+        "q3": np.array([15.0 + 0.5, 0.0 - 0.5, 0.11]),
+    }
+    for q in qdict:
+        np.testing.assert_array_almost_equal(result[q], expected_positions[q])
+
+
+def test_register_noise_sigma_xy_z_raises_if_trap_depth_is_none():
+    """Trap_depth is None."""
+    with pytest.raises(
+        ValueError, match="'trap_depth' must be greater than zero, not None"
+    ):
+        noise_model = NoiseModel(
+            temperature=1.0,
+            trap_depth=None,
+            trap_waist=1.0,
+            runs=1,
+            samples_per_run=1,
+        )
+        noise_model._register_sigma_xy_z()
```

Comment:
It's not `_register_sigma_xy_z()` that's raising the error, it's `NoiseModel`. You can probably delete this test